### DISCUSSION TITLE
About openai-dotnet sdk

### DISCUSSION THREAD

**Discussion by GreenShadeZhang** [microsoft/semantic-kernel#7591] [General]
Does the semantic kernel have plans to introduce openai's dotnet sdk separately to support openai? When I tested using the current semantic kernel code to call some other large language models compatible with the openai api, failures occurred, but everything was normal when using openapi-dotnet alone, so I would like to know whether this separate support method can be used for later version updates.

**Comment by dmytrostruk** [comment]
@GreenShadeZhang We are currently working on transitioning to OpenAI .NET SDK in this feature branch:
https://github.com/microsoft/semantic-kernel/tree/feature-connectors-openai

We also have a migration guide:
https://github.com/microsoft/semantic-kernel/blob/feature-connectors-openai/dotnet/docs/OPENAI-CONNECTOR-MIGRATION.md

It would be great if you could share which model you are trying to use with OpenAI connector, so we can verify your case whether it works with OpenAI .NET SDK connector. Thanks!

**Reply by GreenShadeZhang** [reply]
> @GreenShadeZhang We are currently working on transitioning to OpenAI .NET SDK in this feature branch: https://github.com/microsoft/semantic-kernel/tree/feature-connectors-openai
> 
> We also have a migration guide: https://github.com/microsoft/semantic-kernel/blob/feature-connectors-openai/dotnet/docs/OPENAI-CONNECTOR-MIGRATION.md
> 
> It would be great if you could share which model you are trying to use with OpenAI connector, so we can verify your case whether it works with OpenAI .NET SDK connector. Thanks!


This is my test code. I changed the request URL to the overall path and found that it can be used normally. It should be that there are some problems with the spliced ​​URL when the semantic kernel performs URL conversion.

https://github.com/GreenShadeZhang/BingSearchSummary/blob/master/src/BingSearchSummary.WinUI/App.xaml.cs

```csharp

            var builder = Kernel.CreateBuilder();

            builder.AddOpenAIChatCompletion(modelId: "GLM-4-Air", apiKey: "test", httpClient: GetProxyClient("https://open.bigmodel.cn/api/paas/v4/chat/completions"));

            builder.Plugins.AddFromType<TimeInformationPlugin>();

            services.AddSingleton(builder.Build());



    private static HttpClient GetProxyClient(string baseUrl)
    {
        var httpClient = new HttpClient(new ProxyOpenAIHttpClientHandler(baseUrl));
        //httpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("text/event-stream"));
        return httpClient;
    }
```
If I have some time, I will try your future versions. I think it would be best if I can just replace the endpoint, otherwise I have to replace the entire URL.

The model I use is a local model in China.
https://chatglm.com/
